lib/main.dart

```diff
@@ -11,115 +11,110 @@ class MyApp extends StatelessWidget {
-        // in the middle of the parent.
+      body: Padding(
+        padding: const EdgeInsets.symmetric(horizontal: 16.0),
         child: Column(
-          // Column is also a layout widget. It takes a list of children and
-          // arranges them vertically. By default, it sizes itself to fit its
-          // children horizontally, and tries to be as tall as its parent.
-          //
-          // Column has various properties to control how it sizes itself and
-          // how it positions its children. Here we use mainAxisAlignment to
-          // center the children vertically; the main axis here is the vertical
-          // axis because Columns are vertical (the cross axis would be
-          // horizontal).
-          //
-          // TRY THIS: Invoke "debug painting" (choose the "Toggle Debug Paint"
-          // action in the IDE, or press "p" in the console), to see the
-          // wireframe for each widget.
-          mainAxisAlignment: MainAxisAlignment.center,
-          children: <Widget>[
-            const Text(
-              'You have pushed the button this many times:',
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: [
+            Expanded(
+              child: ListView.builder(
+                controller: scrollController,
+                itemCount: messages.length,
+                reverse: true,
+                padding: const EdgeInsets.symmetric(vertical: 8.0),
+                itemBuilder: (context, i) {
+                  var timestamp = DateTime.fromMillisecondsSinceEpoch(
+                      messages[i]['timestamp']);
+                  return ListTile(
+                    leading: DefaultTextStyle.merge(
+                      style: const TextStyle(color: Colors.indigo),
+                      child: Text('${timestamp.hour}:${timestamp.minute}'),
+                    ),
+                    title: Text('${messages[i]['message']}'),
+                  );
+                },
+              ),
             ),
-            Text(
-              '$_counter',
-              style: Theme.of(context).textTheme.headlineMedium,
+            Padding(
+              padding: const EdgeInsets.symmetric(vertical: 8.0),
+              child: TextField(
+                controller: textMessageController,
+                decoration: InputDecoration(
+                  suffixIcon: IconButton(
+                    icon: const Icon(Icons.send),
+                    onPressed: _sendMessage,
+                  ),
+                  border: const OutlineInputBorder(),
+                  labelText: 'Enter your message',
+                  errorText:
+                      _isTextValidated ? null : 'Message can not be empty',
+                ),
+                onChanged: (value) {
+                  setState(() {
+                    _isTextValidated = true;
+                  });
+                },
+                onSubmitted: (value) => _sendMessage(),
```

Comment:
maybe just pass value as a param instead of using the controller to simplify the code?

Reply:
we need controller to clear the text.

Reply:
oh. I got it. no problem.